src/liberty/devCommands.ts

```diff
@@ -173,10 +173,18 @@ function showListOfPathsToAdd(uris: string[]) {
         if (!selection) {
             return;
         }
-        const result = await projectProvider.addUserSelectedPath(selection, projectProvider.getProjects());
-        const message = localize(`add.project.manually.message.${result}`, selection);
-        (result !== 0) ? console.error(message) : console.info(message); projectProvider.fireChangeEvent();
-        vscode.window.showInformationMessage(message);
+        if (projectProvider.isUntitledWorkspace()) {
+            /**
+             * Saving the selected project to globalstate for adding it to the dashboard after 
+             * reinitialization of the extension when workspace is saved
+             */
+            await projectProvider.getContext().globalState.update('selectedProject', selection);
```

Comment:
One of the things discussed with this issue was the limitation of adding more than one project to the dashboard when a user saves the workspace. This is not required for this PR, but would it not possible to use an array here instead of just a string object? Then all added projects can be saved to the workspace.

Reply:
Yes we can use an array/Set here , but then we would need to handle few cases 
where the user might add/remove projects from dashboard and update the set accordingly , or after making all the changes the user can close the workspace/IDE without saving . and we should clear the data from global state once a Vs code session is closed (Both closing the IDE and workspace scenarios), else this data will be available for other vs code sessions / workspaces and there are chances for wrong projects getting added in different vscode workspaces  .
 Since we have not yet found a way to hook into the flow when user chooses file->close workspace we might not be able to clear the global state here ,  i think it would be good to make these changes ( using a set or an array here and handling the above cases ) along with #386

Reply:
Can you open an enhancement issue to address these scenarios?

Reply:
Sure @TrevCraw